stack.yaml

```diff
@@ -0,0 +1,61 @@
+resolver: lts-5.13
```

Comment:
Hmm, yes, I was worried about having to keep this file up to date.  It's not so much that it would be annoying---I think we would simply forget to do it.

Having a commit of this repo never break can be seen as both good and bad.  Good since it makes it easier for contributors; bad since I *want* to know when things break!  In particular this will be running every night on a CI server I have set up.  So at the very least we do still need a way to say "build with the most recent HEAD versions of everything".

Personally, I lean towards the simplest solution---to list the packages in the `packages` section and require the dependencies to exist locally.  Another benefit of this approach is that it decouples the decision about which versions of the dependencies to use from the `stack.yaml` file itself.  If you want to build with HEAD of everything, just clone all the HEADs.  If you want to build with specific frozen versions of things, just make sure you have those checked out (perhaps helped by some sort of script).  We could make a script to assist contributors in cloning all the appropriate dependencies.

As for the other potential solutions: I really don't want to touch git submodules with a 10-foot pole.  A script to generate `stack.yaml` could certainly work though.